woocommerce/woocommerce-gateway-stripe
# Fix errors when trying to use a 3DS enabled card when Stripe Link is enabled

Fixes #3046

## Changes proposed in this Pull Request:

For stores with Stripe Link enabled, if a customer tries to use a 3D Secure enabled payment method, they would received this error.

<p align="center">
<img width="688" alt="Screenshot 2024-07-24 at 10 32 53 AM" src="https://github.com/user-attachments/assets/534a0f8e-b3e0-41c7-a692-1174aeb19c5a">
</p> 

This error is caused because when we attempt to confirm the payment and display the 3DS challenge modal, we used Stripe API functions [`confirmCardSetup()`](https://docs.stripe.com/js/setup_intents/confirm_card_setup) and  [`confirmCardPayment()`](https://docs.stripe.com/js/payment_intents/confirm_payment). That function is strictly limited to confirming card payments, and doesn't work when confirming Link payment methods. 

To fix this, this PR uses the generic Stripe API function that is capable of confirming card and Link payment methods, instead of the card one. 

## Testing instructions

1. Enable Stripe Link from your payment method settings. 
2. Add a product to your cart.
3. On checkout login to Link by clicking on the Link logo in the email field. 
4. Enter a 3D Secure enabled payment method. eg `4000002760003184`

<p align="center">
<img width="700" alt="Screenshot 2024-07-24 at 10 44 09 AM" src="https://github.com/user-attachments/assets/ded3ea18-eec8-4be0-8ede-45fecd436eb4">
</p> 

5. Attempt to place the order. 
   - On `develop` you will see the error I mentioned above.
   - On this this branch the 3DS modal should be shown as expected. Failing or completing the challenge will display a warning or complete the payment respectively. 

6. Given this impacts other confirmations, I also tested this with Stripe Link **disabled** to confirm 3DS cards confirmations without Link still work. 

> [!Note]
> You will need to run `npm run build` when checking out this branch as it involves JS changes. 





---


**Post merge**
